src/client/app/services/firebase-auth.service.js

```diff
@@ -1,48 +1,54 @@
-(function() {
+(function () {
   "use strict";
 
-  angular.module('app.core')
+  angular.module('app.svc')
 
-  .service('FirebaseAuth', ['FirebaseRef', '$firebaseAuth', '$location', FirebaseAuthService]);
+    .service('FirebaseAuth', ['Profile', '$firebaseAuth', FirebaseAuthService]);
 
-  function FirebaseAuthService(FirebaseRef, $firebaseAuth, $location) {
-    return angular.merge($firebaseAuth(), {
-      authenticate: authenticate,
-      logout: logout
-    });
+  function FirebaseAuthService(Profile, $firebaseAuth) {
 
     /**
      * Authenticate a user with the specified credentials.
      * @param credentials app.Credentials
      * @return Promise
      */
     function authenticate(credentials) {
+      // TODO - Login using Google OAuth, making sure to return the Promise [AUTH-1]
       return $firebaseAuth()
         .$signInWithPopup("google")
-        .then(function(userCredential) {
-          return processUserCredential(userCredential);
-        });
+        .then(processUserCredential);
+      // TODO - Call processUserCredential upon successful login, returning the Promise [AUTH-2]
     }
 
     /**
      * Performs a Firebase logout and returns a promise.
      * @return Promise
      */
     function logout() {
-      // Signout with Firebase and return the promise
-      return $firebaseAuth().$signOut().then(function() {
-        $location.url('/login');
-      }).catch(function(error) {
-        $location.url('/login');
-      });
+      // TODO - Logout, making sure to return the Promise [AUTH-3]
+      return $firebaseAuth().$signOut();
     }
 
+
```

Comment:
Nice, i like it

Reply:
I've updated connect and express. Chat now clears the input field upon submit. Disabled chat when not logged in.

You'll still see errors in the console when logging out. I spent too much time on that rabbit hole already.

Reply:
I've also updated the comments as requested.